services/orchest-api/app/app/core/pipelines.py

```
@@ -806,63 +470,332 @@ def remove_containerization_resources_on_docker(self, task_id, run_config):
+        },
+    }
+    return task
+
+
+def pipeline_to_workflow_manifest(
+    workflow_name: str, pipeline: Pipeline, run_config: Dict[str, Any]
+) -> dict:
+    manifest = {
+        "apiVersion": "argoproj.io/v1alpha1",
+        "kind": "Workflow",
+        "metadata": {"name": workflow_name},
+        "spec": {
+            "entrypoint": "pipeline",
+            "volumes": get_orchest_volumes(
+                host_user_dir=run_config["host_user_dir"],
+                host_project_dir=run_config["project_dir"],
+                host_pipeline_file=os.path.join(
+                    run_config["project_dir"], run_config["pipeline_path"]
+                ),
+            ),
+            # The celery task actually takes care of deleting the
+            # workflow, this is just a failsafe.
+            "ttlStrategy": {
+                "secondsAfterCompletion": 1000,
+                "secondsAfterSuccess": 1000,
+                "secondsAfterFailure": 1000,
+            },
+            "dnsPolicy": "ClusterFirst",
+            "restartPolicy": "Never",
+            # The first entry of this list is the definition of the DAG,
+            # while the second entry is the step definition.
+            "templates": [
+                {
+                    "name": "pipeline",
+                    "retryStrategy": {"limit": "0", "backoff": {"maxDuration": "0s"}},
+                    "dag": {
+                        "failFast": True,
+                        "tasks": [
+                            step_to_workflow_manifest_task(step, run_config)
+                            for step in pipeline.steps
+                        ],
+                    },
+                },
+                {
+                    "name": "step",
+                    "securityContext": {
+                        "runAsUser": 0,
+                        "runAsGroup": int(os.environ.get("ORCHEST_HOST_GID")),
+                        "fsGroup": int(os.environ.get("ORCHEST_HOST_GID")),
```

Not sure whether we want to let it fail or provide a default value to make sure the conversion to `int` always works.

---

I have no strong feeling about this, a slight preference over letting it fail because fixing permissions once this is shipped can be more annoying to users